src/compiler/gcc.rs

```diff
@@ -72,91 +72,94 @@ impl CCompilerImpl for GCC {
     }
 }
 
-#[derive(Clone, Debug)]
-pub enum GCCArgAttribute {
-    TooHard,
-    PassThrough,
-    PreprocessorArgument,
-    DoCompilation,
-    Output,
-    NeedDepTarget,
-    DepTarget,
-    Language,
-    SplitDwarf,
-    ProfileGenerate,
-    TestCoverage,
-    Coverage,
+ArgData!{ pub
+    TooHardFlag(()),
+    TooHard(OsString),
+    PassThrough(OsString),
+    PassThroughPath(PathBuf),
+    PreprocessorArgumentFlag(()),
+    PreprocessorArgument(OsString),
+    PreprocessorArgumentPath(PathBuf),
+    DoCompilation(()),
+    Output(PathBuf),
+    NeedDepTarget(()),
+    DepTarget(OsString),
+    Language(OsString),
+    SplitDwarf(()),
+    ProfileGenerate(()),
+    TestCoverage(()),
+    Coverage(()),
```

Comment:
I don't know if it's feasible, but it would be nicer if the macro could take `$x:ident` without a type and handle it as `()`.

Reply:
To answer my own question: I don't think it's possible with current stable Rust. [RFC 2298](https://github.com/rust-lang/rfcs/blob/master/text/2298-macro-at-most-once-rep.md) adds a `?` repetition operator to macros which would make this feasible. This code works on Nightly (although I couldn't get it to work on the playground for some reason):
https://gist.github.com/luser/0284cecdec8f7a2f789e37ffc52f963e

Reply:
Oh, it needed Edition: 2018 to work on the playground. (My local nightly is a few months out of date.)